ironfish-cli/src/commands/wallet/start.ts

```diff
@@ -0,0 +1,183 @@
+    }),
+    networkId: Flags.integer({
+      char: 'i',
+      default: undefined,
+      description: 'Network ID of an official Iron Fish network to connect to',
+    }),
+    customNetwork: Flags.string({
+      char: 'c',
+      default: undefined,
+      description:
+        'Path to a JSON file containing the network definition of a custom network to connect to',
+    }),
+  }
+
+  node: IronfishNode | null = null
+
+  startDonePromise: Promise<void> | null = null
+
+  async start(): Promise<void> {
+    const [startDonePromise, startDoneResolve] = PromiseUtils.split<void>()
+    this.startDonePromise = startDonePromise
+
+    const { flags } = await this.parse(WalletStart)
+    const { name, workers, upgrade, networkId, customNetwork } = flags
+
+    if (workers !== undefined && workers !== this.sdk.config.get('nodeWorkers')) {
+      this.sdk.config.setOverride('nodeWorkers', workers)
+    }
+
+    if (name !== undefined && name.trim() !== this.sdk.config.get('nodeName')) {
+      this.sdk.config.setOverride('nodeName', name.trim())
+    }
+
+    if (upgrade !== undefined && upgrade !== this.sdk.config.get('databaseMigrate')) {
+      this.sdk.config.setOverride('databaseMigrate', upgrade)
+    }
+
+    if (networkId !== undefined && customNetwork !== undefined) {
+      throw new Error(
+        'Cannot specify both the networkId and customNetwork flags at the same time',
+      )
+    }
+
+    if (networkId !== undefined && networkId !== this.sdk.config.get('networkId')) {
+      this.sdk.config.setOverride('networkId', networkId)
+    }
+
+    if (customNetwork !== undefined && customNetwork !== this.sdk.config.get('customNetwork')) {
+      this.sdk.config.setOverride('customNetwork', customNetwork)
+    }
+
+    const node = await this.sdk.walletNode()
+    const nodeName = this.sdk.config.get('nodeName').trim() || null
+
+    this.log(`\n${ONE_FISH_IMAGE}`)
+    this.log(`Version             ${node.pkg.version} @ ${node.pkg.git}`)
+    this.log(`Wallet Node Name    ${nodeName || 'NONE'}`)
+    if (inspector.url()) {
+      this.log(`Inspector           ${String(inspector.url())}`)
+    }
+    this.log(` `)
+
+    await NodeUtils.waitForOpen(node, () => this.closing)
+
+    if (this.closing) {
+      return startDoneResolve()
+    }
+
+    if (node.internal.get('isFirstRun')) {
+      await this.firstRun(node)
+    }
+
+    await node.start()
+    this.node = node
+
+    startDoneResolve()
+    this.listenForSignals()
+    await node.waitForShutdown()
+  }
+
+  async closeFromSignal(signal: SIGNALS): Promise<void> {
+    this.log(`Shutting down node after ${signal}`)
+    await this.startDonePromise
+    await this.node?.shutdown()
+    await this.node?.closeDB()
+  }
+
+  private async firstRun(node: IronfishNode): Promise<void> {
+    this.log('')
+    this.log('Thank you for installing the Iron Fish Wallet Node.')
+
+    if (!node.config.get(ENABLE_TELEMETRY_CONFIG_KEY)) {
```

Comment:
I think we can remove telemetry stuff since it isn't used in wallet

Reply:
Discussed in huddle. Alternative options included a separate first run flag for the wallet, but a concern was if we run the standalone wallet and then the full node. This will throw an error because the wallet head may not be in the chain and may require a rescan/reset. We'll leave this for now.